# MILES: Multiple-Instance Learning via Embedded Instance Selection
Python implementation of Multiple-Instance Learning via Embedded Instance Selection (MILES) [1] applied to data from a metagenome-wide association study of gut microbiota in type 2 diabetes [2].

This work was done as part of a class project at George Mason University in CS 584 under Professor Huzefa Rangwala. The goal of this project is to efficiently and effectively predict patient phenotype (disease), in this case type 2 diabetes (T2D), from intestinal microbiota of patients using MILES. We are motivated to do this since recent research suggests that factors outside of an organism's genome may contribute to the risk of developing type 2 diabetes. Primary contributors to this project are Kyle Jackson and Robert Truong.

This project follows closely to the work done in [3], with much of the preprocessing already being done by the authors of [3].

# References
[1] Chen, Y., Bi, J., & Wang, J. Z. (2006). MILES: Multiple-instance learning via embedded instance selection. IEEE Transactions on Pattern Analysis and Machine Intelligence, 28(12), 1931-1947.

[2] Qin, J., Li, Y., Cai, Z., Li, S., Zhu, J., Zhang, F., ... & Peng, Y. (2012). A metagenome-wide association study of gut microbiota in type 2 diabetes. Nature, 490(7418), 55-60.

[3] N. LaPierre et al., “CAMIL: Clustering and Assembly with Multiple Instance Learning for Phenotype Prediction,” in IEEE International Conference on Bioinformatics & Biomedicine, Shenzhen, China, 2016.
